pkg/ccl/streamingccl/event.go

```diff
@@ -95,66 +103,83 @@ func (sste sstableEvent) GetSSTable() *roachpb.RangeFeedSSTable {
 	return &sste.sst
 }
 
+// GetDeleteRange implements the Event interface.
+func (sste sstableEvent) GetDeleteRange() *roachpb.RangeFeedDeleteRange {
+	return nil
+}
+
 // GetResolvedSpans implements the Event interface.
 func (sste sstableEvent) GetResolvedSpans() *[]jobspb.ResolvedSpan {
 	return nil
 }
 
 var _ Event = sstableEvent{}
 
-// checkpointEvent indicates that the stream has emitted every change for all
-// keys in the span it is responsible for up until this timestamp.
-type checkpointEvent struct {
-	resolvedSpans []jobspb.ResolvedSpan
+// delRangeEvent is a DeleteRange event that needs to be ingested.
+type delRangeEvent struct {
+	delRange roachpb.RangeFeedDeleteRange
 }
 
-var _ Event = checkpointEvent{}
-
 // Type implements the Event interface.
-func (ce checkpointEvent) Type() EventType {
-	return CheckpointEvent
+func (dre delRangeEvent) Type() EventType {
+	return DeleteRangeEvent
 }
 
 // GetKV implements the Event interface.
-func (ce checkpointEvent) GetKV() *roachpb.KeyValue {
+func (dre delRangeEvent) GetKV() *roachpb.KeyValue {
```

Comment:
I think this really highlights the fact that having generic Event type is not helping.
every implementation has all but 1 method returning nil;

I think Event could be just a concrete type (a union).
Okay to leave a TODO for a cleanup later.